OpenRA.Mods.Common/Widgets/Logic/Ingame/OrderButtonsChromeLogic.cs

```diff
@@ -47,20 +49,25 @@ public PowerdownOrderButtonLogic(Widget widget, World world)
 
 	public class BeaconOrderButtonLogic : ChromeLogic
 	{
+		static string beaconcursor;
+		public static Action<BeaconOrderGenerator> AssignBeaconCursor = (orderGenerator) => orderGenerator.BeaconCursor = beaconcursor;
+
 		[ObjectCreator.UseCtor]
 		public BeaconOrderButtonLogic(Widget widget, World world)
 		{
+			beaconcursor ??= world.LocalPlayer.PlayerActor.Info.TraitInfo<PlaceBeaconInfo>().BeaconCursor;
+
 			if (widget is ButtonWidget beacon)
-				OrderButtonsChromeUtils.BindOrderButton<BeaconOrderGenerator>(world, beacon, "beacon");
+				OrderButtonsChromeUtils.BindOrderButton(world, beacon, "beacon", AssignBeaconCursor);
```

Comment:
You can inline the lambda definition here

```suggestion
				OrderButtonsChromeUtils.BindOrderButton(world, beacon, "beacon", og => og.BeaconCursor = beaconCursor);
```